### PR TITLE
Update conformance test walker to read table based conformance tests

### LINKED ISSUE #68339: Failure to promote non-string-literal e2e tests to Conformance
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind feature

**What happened**:
Table-based e2e tests are not getting updated to [golden conformance list](https://github.com/kubernetes/kubernetes/blob/master/test/conformance/testdata/conformance.txt) leading to the following error- ( mentioned in https://github.com/kubernetes/kubernetes/pull/67612#issuecomment-418616392)
```
INFO: From Listing all conformance tests. //test/conformance:list_conformance_tests:
ERROR at test/e2e/common/runtime.go:232:29: framework.ConformanceIt() called with non-literal argument
ERROR: non-literal argument &{0xc42038d2e0 9790 [0xc42038d300 0xc42038d360] 0 9858} at test/e2e/common/runtime.go:232:29
ERROR at test/e2e/common/runtime.go:355:29: framework.ConformanceIt() called with non-literal argument
ERROR: non-literal argument &{0xc42038b7a0 17397 + 0xc42038b7c0} at test/e2e/common/runtime.go:355:29
Target //test/conformance:list_conformance_tests up-to-date:
  bazel-genfiles/test/conformance/conformance.txt
INFO: Elapsed time: 11.206s, Critical Path: 10.55s
```
framework.ConformanceIt() requires the first parameter to be `string literal` and [table based e2e tests](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/common/runtime.go#L204) are associated with `string formation` at runtime based on different cases/test-scenarios.

**What you expected to happen**:
Either disapprove the use of table-based e2e tests or suggestion on an alternative approach to add such tests to Conformance.

**How to reproduce it (as minimally and precisely as possible):**
1. Add framework.ConformanceIt() to [table-based e2e test](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/common/runtime.go#L204).
2. Run ```bazel build //test/conformance:list_conformance_tests```

/area conformance
@kubernetes/sig-testing
cc @brahmaroutu

### PR DESCRIPTION
**Pull request by mgdevstack**
**What this PR does / why we need it**:
- Allows adding table based conformance tests to "conformance golden list"
- Conformance comment added above `framework.ConformanceIt()` block is updated in Conformance Doc.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #68339

**Special notes for your reviewer**:
Currently, same comment block is added to all table based tests.
We can further improve codebase to parse test specific conformance data from table based test like below. (if it make sense to put this effort, embedding conformanceComment in testCase options)
```
for _, testCase := range []struct {
    name      string
    container v1.Container
    phase     v1.PodPhase
    message   gomegatypes.GomegaMatcher
    comment   conformanceComment
}{
    {
        name: "if TerminationMessagePath is set [NodeConformance]",
        container: v1.Container{
            .........
            },
        },
        phase:   v1.PodSucceeded,
        message: Equal("DONE"),
        comment: conformanceComment{
            name:        "Container Runtime, Termination Message, TerminationMessagePath set",
            description: "Container is created with custom terminationMessagePath as Root user and container is voluntarily terminated. Termination Message MUST match DONE and terminationMessagePath MUST has Termination Message matching content",
        },
    },
} {

    /*
        Release : v1.13
        Testname: testCase.comment.name
        Description: testCase.comment.description
    */
    framework.ConformanceIt(fmt.Sprintf("should report termination message %s", testCase.name), func() {})
}
```
**Release note**:

```release-note
None
```
/area conformance
cc @brahmaroutu

### CODE REVIEW

**Comment by mgdevstack** [comment]
Holding this PR to do refactoring and follow standard naming convention.
/hold

**Comment by mgdevstack** [comment]
kind/feature

File: test/conformance/walk.go [hunk 1]
```diff
@@ -74,39 +83,43 @@ type conformanceData struct {
...
+	case *ast.BasicLit, *ast.CallExpr, *ast.BinaryExpr:
+		comment := v.comment(at)
+		pos := v.FileSet.Position(at.Pos())
+		cd.URL = fmt.Sprintf("%s%s#L%d", *baseURL, pos.Filename, pos.Line)
+
+		lines := strings.Split(comment, "\n")
+		cd.Description = ""
+		for _, line := range lines {
+			line = strings.TrimSpace(line)
+			if strings.HasPrefix(line, "Testname:") {
```
Comment (cheftako) [inline test/conformance/walk.go]: Seems like "Testname:" and "Description:" should be const. Then we can do things like call len() on this for things like indexing and the code is both more clear and easier to maintain.

File: test/conformance/walk.go [hunk 2]
```diff
@@ -254,7 +486,7 @@ func (v *visitor) firstArg(n *ast.CallExpr) string {
 	if ident, ok := n.Args[0].(*ast.Ident); ok {
 		return ident.String()
 	}
-	return "*"
+	return ""
```
Comment (cheftako) [inline test/conformance/walk.go]: Can we make sure this is consistent with the comment describing what the function returns?

Comment (mgdevstack) [reply test/conformance/walk.go]: Thanks for pointing. The codebase is updated as requested. Though I did not fully understand the benefit of replacing %s/%d with "*" from the first arg of all "Describe|KubeDescribe|SIGDescribe|Context" if arg[0] is non-string literal.

File: test/conformance/walk.go (hunk 1 above)
Comment (mgdevstack) [reply test/conformance/walk.go]: Done.

**Comment by mgdevstack** [comment]
It would be great if we can focus on including test scenarios/options inside single ConformanceIt() block [like this](https://github.com/kubernetes/kubernetes/blob/85a8f97cb14a908aefc7ac0890f0cd87f1ea46f8/test/e2e/auth/service_accounts.go#L247-L411) instead of [iterating over range](https://github.com/kubernetes/kubernetes/blob/85a8f97cb14a908aefc7ac0890f0cd87f1ea46f8/test/e2e/auth/service_accounts.go#L247-L411).

**Comment by mgdevstack** [comment]
We can close this PR as we are intended to convert all table based tests into individual tests. If in future we need to upgrade this test walker, we can think to cherry pick this commit.

**Feature in this PR:**
We can include ConformanceComment block in list of tests data struct and then all variables can be interpolated to respective values. (as mentioned code-block in PR description)

**Comment by mgdevstack** [comment]
/close
PR seems not considerable at this moment.

**Pull request closed without merging**